modeling/simulator/theoretical-simulator.go

```diff
@@ -146,6 +147,19 @@ func (zd zoneSGDetails) getZoneToZoneTraffic(region types.RegionInfo, endpointSl
 
 // calculate simulation result based on probabilities
 func getSimulationResult(zd zoneSGDetails, region types.RegionInfo, endpointSlices map[string]types.EndpointSliceGroup, zoneTrafficToZone map[string]map[string]float64) types.SimulationResult {
+	// check unexpected behaviors
+	for zone := range region.ZoneDetails {
```

Comment:
You can update this loop to get the values of the map instead of the key. This should mean that you can replace all of your `zd[zone]` references below with just `zone`.
```suggestion
	for _, zone := range region.ZoneDetails {
```

Reply:
Its not the same structure that is used below. This is meant to traverse all zones in zoneSGDetails.